src/common/drain/drain.js

```diff
@@ -6,33 +6,66 @@
 import { ee } from '../event-emitter/contextual-ee'
 import { mapOwn } from '../util/map-own'
 import { registerHandler as defaultRegister } from '../event-emitter/register-handler'
+import { featurePriority } from '../../loaders/features/features'
 
+const registry = {}
 
-// calls will need to update to call this more directly so we can explicitly pass in the ee and handler
-export function drain(agentIdentifier, group = 'feature') {
-  const baseEE = agentIdentifier ? ee.get(agentIdentifier) : ee
-  const handlers = defaultRegister.handlers
-  if (!baseEE.backlog || !handlers) return
+export function registerDrain(agentIdentifier, group) {
+  const item = { staged: false, priority: featurePriority[group] || 0 }
+  curateRegistry(agentIdentifier)
+  if (!registry[agentIdentifier].get(group)) registry[agentIdentifier].set(group, item)
+}
 
-  var bufferedEventsInGroup = baseEE.backlog[group]
-  var groupHandlers = handlers[group]
-  if (groupHandlers) {
-    // don't cache length, buffer can grow while processing
-    for (var i = 0; bufferedEventsInGroup && i < bufferedEventsInGroup.length; ++i) { // eslint-disable-line no-unmodified-loop-condition
-      emitEvent(bufferedEventsInGroup[i], groupHandlers)
-    }
+function curateRegistry(agentIdentifier) {
+  if (!agentIdentifier) return
+  if (!registry[agentIdentifier]) registry[agentIdentifier] = new Map()
+}
 
-    mapOwn(groupHandlers, function (eventType, handlerRegistrationList) {
-      mapOwn(handlerRegistrationList, function (i, registration) {
-        // registration is an array of: [targetEE, eventHandler]
-        registration[0].on(eventType, registration[1])
-      })
+/**
+ * Drain buffered events out of the EE. Each feature **should** have its events bucket into a "group".  The feature should drain this group explicitly
+ * @param {string} agentIdentifier 
+ * @param {string} group 
+ */
+export function drain(agentIdentifier = "", featureName = 'feature') {
+  curateRegistry(agentIdentifier)
+  // if its not in the registry, that means the instrument file was bypassed.  This could happen in tests, or loaders that directly import the agg
+  if (!agentIdentifier || !registry[agentIdentifier].get(featureName)) return drainGroup(featureName)
+  registry[agentIdentifier].get(featureName).staged = true
+
+  const items = Array.from(registry[agentIdentifier])
+  if (items.every(([key, values]) => values.staged)) {
+    items.sort((a, b) => a[1].priority - b[1].priority)
```

Comment:
Its unclear whether the order of features drained has consequences or not, so manually enforcing this for now to match legacy build order.

Reply:
Most likely not, since they get buffered into EE then sent out on payloads that DEM don't mandate an ordering or statefulness to. But meh.